### ISSUE TITLE
`shap.Explanation` lacks documentation

### ISSUE THREAD

**Issue by banderlog** [shap/shap#1956]
The latest docs: https://shap.readthedocs.io/en/latest/generated/shap.Explanation.html#shap.Explanation

All attributes are empty.

+ `data` is just copy of data on which explainer was called?
+ `values` are actual shap values?
+ `base_values` what are they?

**Comment by ky66** [comment]
Yeah I had the same problem. There is not adequate explanation for these. And model-specific functions such as DeepExplainer have the situation where when .shap_values is called on them, they generate only the values not the base and data. I have been trying to make a beeswarm plot with DeepExplainer but I could not do it. .shap_values works with everything else.

**Comment by Data-drone** [comment]
From what I can figure yes:

`data` is for the model features
`values` are the shap values
`base_values` I thought was just the target value but this https://towardsdatascience.com/explain-machine-learning-models-using-shap-library-e05a1583c34f suggests that it is the mean of the target?

**Comment by banderlog** [comment]
Now all marked as "Pass-through from the underlying slicer object.", no link on that "slicer object" description.

The word "slicer"  has only two mentions in the docs, and the other one is a name of python package: https://pypi.org/project/slicer/

There are no mention of "base_values" in the https://github.com/interpretml/slicer

**Comment by Saloni-0465** [comment]
Hi, I’d like to work on improving the documentation for shap.Explanation attributes (values, base_values, data). Is this still open for contributions?

**Comment by ashwinkothavade** [comment]
Opened a PR for this. Rewrote the class docstring to actually describe what `Explanation` is, documented every `__init__` parameter with shape and semantics, replaced the "Pass-through from the underlying slicer object" boilerplate on the attribute properties with substantive descriptions of `values`, `base_values`, `data`, `display_data`, `feature_names`, `main_effects` etc., and added an Examples block. ReadTheDocs preview should render on the PR once CI kicks off.
